.github/workflows/link-check-deploy.yml

```diff
@@ -0,0 +1,48 @@
+name: Check new links against deployment
+# This workflow "triggers" and skips on deployment because GitHub Actions /
+# Checks refuses to show the check on deployment_status
+on:
+  - deployment
+  - deployment_status
+jobs:
+  run:
+    name: Initialize
+    runs-on: ubuntu-latest
+    if:
+      github.event.deployment.ref != 'main' &&
+      github.event.deployment_status.state == 'success'
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - id: build_check
+        uses: LouisBrunner/checks-action@v1.0.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          name: Report
+          status: queued
+      - name: Run Link Check
+        id: check
+        uses: iterative/link-check.action@v0.11
+        with:
+          diff: main
+          configFile: config/link-check/config.yml
+          rootURL: '${{ github.event.deployment.payload.web_url }}'
+          output: checksAction
+      - uses: LouisBrunner/checks-action@v1.0.0
+        if: ${{ success() }}
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          check_id: ${{ steps.build_check.outputs.check_id }}
+          status: completed
+          conclusion: ${{ steps.check.outputs.conclusion }}
+          output: ${{ steps.check.outputs.output }}
```

Comment:
Is there a way to name the report check? This is pretty unclear that's it's checking links 🤔 

<img width="423" alt="image" src="https://user-images.githubusercontent.com/43496356/172842962-aed7e739-3452-40d4-b85f-ce340ec1f630.png">

Reply:
I'm not sure what's going on, seems to be a GH Actions bug where this action's report is taking the name of another action. Maybe doing a CLI implementation will solve the issue by making there less moving parts in check creation.